docs/docs/community/release-process.md

```diff
@@ -0,0 +1,75 @@
+---
+id: release-process
+title: The Nebari release process
+description: The Nebari release process and workflow
+---
+
+# The Nebari release process
+
+This page describes the high-level details of cutting a new Nebari release.
+
+|                   |                                                                                                                                                                                                  |
+| :---------------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|      Cadence      |                                                                                             Monthly                                                                                              |
+| Versioning System |                                                                                    [CalVer](#calver-details)                                                                                     |
+| Release Checklist |     [Link to issue template](https://github.com/nebari-dev/nebari/issues/new?assignees=&labels=type%3A+release+%F0%9F%8F%B7&template=release-checklist.md&title=%5BRELEASE%5D+%3Cversion%3E)     |
+| Testing Checklist | [Link to issue template](https://github.com/nebari-dev/nebari/issues/new?assignees=&labels=type%3A+release+%F0%9F%8F%B7&template=testing-checklist.md&title=Testing+checklist+for+%3Cversion%3E) |
+
+## Release Captain responsibilities
+
+For every release, there is an assigned "Release Captain". It is the responsibility of this person to manage both the release and testing checklists as well as to communicate the status of the release to the rest of the Nebari development team and the community.
+
+Communication channels that need to be updated are:
+
+- the release checklist issue itself
+- the `#nebari-dev` slack channel
```

Comment:
Since this is not a public channel and it's likely that folks will get confused I prefer removing this